datalad/runner/nonasyncrunner.py

```diff
@@ -333,7 +333,8 @@ def _locked_run(self) -> dict | _ResultGenerator:
             # enqueue the input that is already provided.
             self.write_stdin = True
             self.stdin_queue = Queue()
-            self.stdin_queue.put(self.stdin)
+            self.stdin_queue.put(self.stdin if isinstance(self.stdin, bytes)
+                                 else self.stdin.encode(encoding='utf-8'))
```

Comment:
Basically correct, but I would use the protocol_kwargs encoding value, if it is set. Besides that, the indentation in this file has a different style and strings here are always delimited with double-quotes.

```suggestion
            self.stdin_queue.put(
                self.stdin
                if isinstance(self.stdin, bytes)
                else self.stdin.encode(
                    encoding=self.protocol_kwargs.get("encoding", "utf-8")))
```